src/test/java/apicurito/tests/configuration/templates/ApicuritoTemplate.java

```diff
@@ -62,38 +66,110 @@ public static void deployUsingTemplate() {
+        log.info("Deploying Service from " + TestConfiguration.apicuritoOperatorServiceUrl());
+        final String output = OpenShiftUtils.binary().execute(
+                "create",
+                "-n", TestConfiguration.openShiftNamespace(),
+                "-f", TestConfiguration.apicuritoOperatorServiceUrl()
+        );
+    }
+
+    private static void deployRole() {
+        log.info("Deploying Role from " + TestConfiguration.apicuritoOperatorRoleUrl());
+        final String output = OpenShiftUtils.binary().execute(
+                "create",
+                "-n", TestConfiguration.openShiftNamespace(),
+                "-f", TestConfiguration.apicuritoOperatorRoleUrl()
+        );
+    }
+
+    private static void deployRoleBinding() {
+        log.info("Deploying Role binding from " + TestConfiguration.apicuritoOperatorRoleBindingUrl());
+        final String output = OpenShiftUtils.binary().execute(
+                "create",
+                "-n", TestConfiguration.openShiftNamespace(),
+                "-f", TestConfiguration.apicuritoOperatorRoleBindingUrl()
+        );
+    }
+
+    private static void deployOperator() {
+        log.info("Deploying operator from " + TestConfiguration.apicuritoOperatorUrl());
+        final String output = OpenShiftUtils.binary().execute(
+                "create",
+                "-n", TestConfiguration.openShiftNamespace(),
+                "-f", TestConfiguration.apicuritoOperatorUrl()
+        );
+    }
+
     public static void cleanNamespace() {
         TestConfiguration.printDivider("Deleting namespace...");
+
         try {
-            OpenShiftUtils.client().apps().statefulSets().inNamespace(TestConfiguration.openShiftNamespace()).delete();
-            OpenShiftUtils.client().extensions().deployments().inNamespace(TestConfiguration.openShiftNamespace()).delete();
-            OpenShiftUtils.client().customResourceDefinitions().delete();
+            //OCP4HACK - openshift-client 4.3.0 isn't supported with OCP4 and can't create/delete templates, following line can be removed later
+            OpenShiftUtils.binary().execute("delete", "template", "--all");
+            OpenShiftUtils.getInstance().apps().statefulSets().inNamespace(TestConfiguration.openShiftNamespace()).delete();
+            //OpenShiftUtils.getInstance().extensions().deployments().inNamespace(TestConfiguration.openShiftNamespace()).delete();
+            OpenShiftUtils.getInstance().roles().delete();
+            OpenShiftUtils.getInstance().roleBindings().delete();
+            OpenShiftUtils.getInstance().customResourceDefinitions().delete();
         } catch (KubernetesClientException ex) {
             // Probably user does not have permissions to delete.. a nice exception will be printed when deploying
         }
-        OpenShiftUtils.getInstance().cleanAndAssert();
+        OpenShiftUtils.getInstance().clean();
+        OpenShiftUtils.getInstance().waiters().isProjectClean().waitFor();
```

Comment:
I'm having difficulties there - the previous steps don't seem to remove the operator deployments on my side.

Reply:
fixed